x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java

```diff
@@ -106,6 +116,19 @@ public Configuration(BlockStreamInput in) throws IOException {
         } else {
             this.queryStartTimeNanos = -1;
         }
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_CONFIGURATION_WITH_FEATURES)) {
+            this.activeEsqlFeatures = in.readCollectionAsImmutableSet(StreamInput::readString);
+        } else {
+            this.activeEsqlFeatures = Set.of();
+        }
```

Comment:
I wonder if we could avoid serializing this. Like, maybe it doesn't belong in the configuration - maybe it's another parameter that we add to `FunctionDefinition.Builder` or something.

I feel like we'd be better off reading from the `Set<NodeFeatures>` that are active in the ctor of the Function and setting a boolean. That feels like it'd be easier to debug and test and the functions could have that `boolean` in their descriptions so even in `profile` we could see what it decided to do. I feel like there aren't going to be *that* many of these functions with behavior changes so the extra booleans here and there are ok. And the "you can see what it chose" behavior is nicer than serializing all of these strings.

Also! I kind of like the idea of choosing at function resolution time more than copying the list of features, saving it, and choosing at function translation time. It amounts to the same thing but it *feels* better in my heart.